src/test/java/com/puppycrawl/tools/checkstyle/internal/utils/TestUtil.java

```diff
@@ -237,4 +251,19 @@ public static int adjustFlushCountForOutputStreamClose(int flushCount) {
         return result;
     }
 
+    /**
+     * Executes a test method in a thread with limited stack size.
+     *
+     * @param executable the method to execute
+     * @see <a href="https://docs.oracle.com/javase/specs/jvms/se15/html/jvms-2.html#jvms-2.5.2">
+     *      JVMS &sect;2.5.2</a>
+     */
+    public static void executeWithLimitedStackSize(Executable executable) {
+        final Thread thread = new Thread(null, () -> {
+            assertDoesNotThrow(executable, "No exception expected");
+        }, "LimitedStackThread", MINIMAL_STACK_SIZE);
+        assertDoesNotThrow(() -> thread.join(10000),
```

Comment:
method name is `executeWithLimitedStackSize` but in fact time is limited as well, since there is a timeout for join.
btw, I wonder is there some library tool for such things instead of creating your own wrapper for parallel execution?

Reply:
`executeWithLimitedStackSizeAndTimeout`

Reply:
done